src/KafkaFlow.Retry.UnitTests/Repositories/SqlServer/Readers/RetryQueueReaderTests.cs

```diff
@@ -0,0 +1,198 @@
+                .Returns(new MessageHeader("key", new byte[2]));
+
+            reader = new RetryQueueReader(
+                retryQueueAdapter.Object,
+                retryQueueItemAdapter.Object,
+                retryQueueItemMessageAdapter.Object,
+                retryQueueItemMessageHeaderAdapter.Object);
+        }
+
+        [Fact]
+        public void RetryQueueReader_Read_Success()
+        {
+            // Arrange
+            var wrapper = new RetryQueuesDboWrapper
+            {
+                QueuesDbos = new[]
+                {
+                    new RetryQueueDbo
+                    {
+                        Id = 1,
+                        CreationDate = DateTime.UtcNow,
+                        LastExecution = DateTime.UtcNow,
+                        Status = RetryQueueStatus.Active,
+                        QueueGroupKey = "1",
+                        SearchGroupKey = "1"
+                    }
+                },
+                ItemsDbos = new[]
+                {
+                    new RetryQueueItemDbo
+                    {
+                        Description = "description",
+                        DomainRetryQueueId = Guid.NewGuid(),
+                        CreationDate = DateTime.UtcNow,
+                        IdDomain = Guid.NewGuid(),
+                        ModifiedStatusDate = DateTime.UtcNow,
+                        AttemptsCount = 1,
+                        Id = 1,
+                        LastExecution = DateTime.UtcNow,
+                        RetryQueueId = 1,
+                        SeverityLevel = Durable.Common.SeverityLevel.High,
+                        Sort = 1,
+                        Status = RetryQueueItemStatus.InRetry
+                    }
+                },
+                MessagesDbos = new[]
+                {
+                    new RetryQueueItemMessageDbo
+                    {
+                        IdRetryQueueItem = 1,
+                        Key = new byte[] { 1, 3 },
+                        Offset = 2,
+                        Partition = 1,
+                        TopicName = "topicName",
+                        UtcTimeStamp = DateTime.UtcNow,
+                        Value = new byte[] { 2, 4, 6 }
+                    }
+                },
+                HeadersDbos = new[]
+                {
+                    new RetryQueueItemMessageHeaderDbo
+                    {
+                        Id = 1,
+                        Key = "key",
+                        Value = new byte[2],
+                        RetryQueueItemMessageId = 1
+                    }
+                },
+            };
+
+            // Act
+            var result = reader.Read(wrapper);
+
+            // Assert
+            result.Should().NotBeEmpty();
+            result.Count.Should().Be(1);
+        }
+
+        [Theory]
+        [MemberData(nameof(DataTest))]
+        public void RetryQueueReader_Read_Validation(
+            object wrapper)
```

Comment:
This argument should be the concrete objects instead of using `object`.

Reply:
done